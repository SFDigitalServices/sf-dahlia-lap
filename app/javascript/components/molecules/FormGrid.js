import React from 'react'

import classNames from 'classnames'

const FormGrid = {}

FormGrid.Row = ({ children, paddingBottom, expand }) => {
  const divClassNames = classNames(
    'form-grid',
    'row',
    {
      expand: expand,
      // Override left and right margin if expand is false.
      // Otherwise default row styling causes negative margins.
      'margin-left-none': !expand,
      'margin-right-none': !expand,
      'padding-bottom': paddingBottom
    }
  )
  return (
    <div className={divClassNames}>
      {children}
    </div>
  )
}

FormGrid.Row.defaultProps = {
  expand: true
}

// Wrapper for each form field.
// NOTE: Grid is hardcoded. This should not be generic element
FormGrid.Item = ({ children, small = false, fullWidth = false }) => (
  <div className={classNames(
    'form-grid_item column',
    { 'small-6': !(small || fullWidth) },
    { 'small-3': small },
    { 'small-12': fullWidth }
  )}>
    {children}
  </div>
)

// Use for form fields that have a shared label.
FormGrid.Group = ({ children, label }) => (
  <div className='form-group'>
    { label && <label htmlFor=''>{label}</label>}
    {children}
  </div>
)

export default FormGrid
