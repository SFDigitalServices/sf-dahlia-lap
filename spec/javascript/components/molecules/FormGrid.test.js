import React from 'react'
import { shallow } from 'enzyme'
import FormGrid from 'components/molecules/FormGrid'

describe('FormGrid.Item', () => {
  test('renders the FormGrid.Item properly with default props', () => {
    const wrapper = shallow(<FormGrid.Item>ABC</FormGrid.Item>)
    expect(wrapper.find('.form-grid_item')).toHaveLength(1)
    expect(wrapper.find('.small-6')).toHaveLength(1)
  })

  test('renders children properly', () => {
    const wrapper = shallow(<FormGrid.Item>ABC</FormGrid.Item>)
    expect(wrapper.text()).toEqual('ABC')
  })

  test('renders the FormGrid.Item properly with width=25%', () => {
    const wrapper = shallow(<FormGrid.Item width='25%'>ABC</FormGrid.Item>)
    expect(wrapper.find('.form-grid_item')).toHaveLength(1)
    expect(wrapper.find('.small-3')).toHaveLength(1)
  })

  test('renders the FormGrid.Item properly with width=33%', () => {
    const wrapper = shallow(<FormGrid.Item width='33%'>ABC</FormGrid.Item>)
    expect(wrapper.find('.form-grid_item')).toHaveLength(1)
    expect(wrapper.find('.small-4')).toHaveLength(1)
  })
})
