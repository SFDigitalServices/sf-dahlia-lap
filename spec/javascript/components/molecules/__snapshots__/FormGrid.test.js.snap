// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGrid.Item should render FormGrid.Item 1`] = `
<div
  className="form-grid_item column small-6 medium-6 large-6"
>
  ABC
</div>
`;

exports[`FormGrid.Item should render small FormGrid.Item 1`] = `
<div
  className="form-grid_item column small-3 medium-3 large-3"
>
  ABC
</div>
`;
